tests/subsys/bluetooth/mesh/scheduler_model/message_validity/src/main.c

```diff
@@ -0,0 +1,553 @@
+
+static void onoff_set(struct bt_mesh_onoff_srv *srv, struct bt_mesh_msg_ctx *ctx,
+		      const struct bt_mesh_onoff_set *set, struct bt_mesh_onoff_status *rsp)
+{
+}
+
+static void onoff_get(struct bt_mesh_onoff_srv *srv, struct bt_mesh_msg_ctx *ctx,
+		      struct bt_mesh_onoff_status *rsp)
+{
+}
+
+struct bt_mesh_onoff_srv_handlers onoff_srv_handlers = {
+	.set = onoff_set,
+	.get = onoff_get,
+};
+
+struct bt_mesh_onoff_srv onoff_srv = BT_MESH_ONOFF_SRV_INIT(&onoff_srv_handlers);
+
+static void srv_action_set_cb(struct bt_mesh_scheduler_srv *srv, struct bt_mesh_msg_ctx *ctx,
+			      uint8_t idx, struct bt_mesh_schedule_entry *entry)
+{
+	ztest_check_expected_value(srv);
+	ztest_check_expected_value(ctx);
+	ztest_check_expected_value(idx);
+	ztest_check_expected_data(entry, sizeof(struct bt_mesh_schedule_entry));
+}
+
+static struct bt_mesh_scheduler_srv sched_srv =
+	BT_MESH_SCHEDULER_SRV_INIT(&srv_action_set_cb, &time_srv);
+
+static void cli_action_status_handler(struct bt_mesh_scheduler_cli *cli,
+				      struct bt_mesh_msg_ctx *ctx, uint8_t idx,
+				      const struct bt_mesh_schedule_entry *action)
+{
+	ztest_check_expected_value(cli->model);
+	ztest_check_expected_value(ctx);
+	ztest_check_expected_value(idx);
+	ztest_check_expected_data(action, sizeof(struct bt_mesh_schedule_entry));
+}
+
+static void cli_status_handler(struct bt_mesh_scheduler_cli *cli, struct bt_mesh_msg_ctx *ctx,
+			       uint16_t schedules)
+{
+	ztest_check_expected_value(cli->model);
+	ztest_check_expected_value(ctx);
+	ztest_check_expected_value(schedules);
+}
+
+static struct bt_mesh_scheduler_cli sched_cli = {
+	.action_status_handler = &cli_action_status_handler,
+	.status_handler = &cli_status_handler,
+};
+
+/** Mocking mesh composition data
+ *
+ * The mocked composition data consists of 1 element with
+ * 1 Scheduler server, 1 Scheduler client, 1 Generic onoff server,
+ * and 1 Scene server.
+ *
+ * |      1     |
+ * |:----------:|
+ * | Sched. Srv |
+ * | Sched. Cli |
+ * | Onoff Srv  |
+ * | Scene Srv  |
+ *
+ */
+static struct bt_mesh_model mock_models_elem[4] = {
+	{
+		.user_data = &sched_srv,
+		.id = BT_MESH_MODEL_ID_SCHEDULER_SRV,
+	},
+	{
+		.user_data = &sched_cli,
+		.id = BT_MESH_MODEL_ID_SCHEDULER_CLI,
+	},
+	{
+		.user_data = &onoff_srv,
+		.id = BT_MESH_MODEL_ID_GEN_ONOFF_SRV,
+	},
+	{
+		.user_data = &scene_srv,
+		.id = BT_MESH_MODEL_ID_SCENE_SRV,
+	},
+};
+
+static struct bt_mesh_elem mock_elem = {
+	.addr = 1,
+	.model_count = ARRAY_SIZE(mock_models_elem),
+	.models = mock_models_elem,
+};
+
+struct mock_comp_data {
+	size_t elem_count;
+	struct bt_mesh_elem *elem;
+} mock_comp = { .elem_count = 1, .elem = &mock_elem };
+
+static struct bt_mesh_model *mock_mod_srv = &mock_models_elem[0];
+static struct bt_mesh_model *mock_mod_cli = &mock_models_elem[1];
+
+/* Redefined mocks */
+
+struct bt_mesh_elem *bt_mesh_model_elem(struct bt_mesh_model *mod)
+{
+	return 0;
```

Comment:
If you actually meant to hardcode this to return NULL, please use `NULL` and not `0`. Unless you meant to return `&mock_comp.elem[0]`